### PR TITLE
Remove wire cutting module

### LINKED ISSUE #361: Remove cutting.wire_cutting package
Remove deprecated package. Was replaced by `circuit_knitting.cutting.cutqc` in v0.2.

### PR DESCRIPTION
**Pull request by caleb-johnson**
Resolves #361 

Wire cutting module was moved to cutqc module in v0.2.0. Let's clean up the old module now that a release has passed.

On hold until v0.4

### CODE REVIEW

File: test/cutting/test_backwards_compatibility.py
[the remark this answers is not in the document]
Comment (caleb-johnson) [reply test/cutting/test_backwards_compatibility.py:1]: The repo is changing rapidly. I don't see any reason to keep things around longer than one release cycle, especially when it was many months between. Giving any deprecation warning at all for an alpha release is more than fine imo

[the remark this answers is not in the document]
Comment (garrison) [reply test/cutting/test_backwards_compatibility.py:1]: It's an unrelated change, and we haven't actually had any reason to break any of the tests in this file (not yet, at least, although we plan to break some minor details in the next release cycle).

[the remark this answers is not in the document]
Comment (caleb-johnson) [reply test/cutting/test_backwards_compatibility.py:1]: I misunderstood your initial comment. Disregard my previous comment, I see your point now :)

File: releasenotes/notes/remove-wire-cutting-ad0e7ab8af699561.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+upgrade:
+  - |
+    Removed the :mod:`circuit_knitting.wire_cutting` module. Users are now expected to use the :mod:`circuit_knitting.cutting.cutqc` module instead.
```
Comment (garrison) [inline releasenotes/notes/remove-wire-cutting-ad0e7ab8af699561.yaml:4]: I think this is perhaps more clear about the fact that no code changes are required; it's only the import location that changed.

```suggestion
    Removed the :mod:`circuit_knitting.wire_cutting` module. Users are now expected to import from the :mod:`circuit_knitting.cutting.cutqc` module instead.
```

**Review by garrison — APPROVED** [review]
I think right after the 0.4.0 release would be my preferred time to do this, in case some users finally upgrade for the first time in a while once 0.4 is out.  But this has been deprecated since the 0.2.0 release on May 30 (which is just over 3 months ago), so technically we are still within Qiskit's guidelines if we do this, and we don't promise to subscribe to their deprecation policy anyway.

**Pull request merged**